packages/back-end/src/api/features/listFeatures.ts

```diff
@@ -24,13 +27,44 @@ export const listFeatures = createApiRequestHandler(listFeaturesValidator)(
       req.query.projectId
     );
 
+    // If SDK clientKey is provided, get the SDK connection and use its projects/environment
+    let filteredFeatures = features;
+    if (req.query.clientKey) {
+      const sdkConnection = await findSDKConnectionByKey(req.query.clientKey);
+      if (!sdkConnection) {
+        throw new Error("Invalid SDK connection key");
+      }
+
+      const payload = await getFeatureDefinitions({
+        context: req.context,
+        capabilities: getConnectionSDKCapabilities(sdkConnection),
+        environment: sdkConnection.environment,
+        projects: sdkConnection.projects,
+        encryptionKey: sdkConnection.encryptPayload
+          ? sdkConnection.encryptionKey
+          : "",
```

Comment:
We should always pass in an empty string (or undefined) here.  Otherwise, the response might be encrypted and `payload.features` will be an empty object.

Reply:
good catch